test/run

```diff
@@ -14,6 +14,11 @@ testCompilePackageNames() {
   assertCaptured "Writing profile script"
   assertCaptured "Rewrite package-config files"
   assertCapturedSuccess
+
+  # Test cache re-use
```

Comment:
@edmorley I'm a bit confused by this. Shouldn't we see "Reusing cache" if we're testing cache re-use? 

Also, do we need a second test to validate invalidation based on stack changes?

Reply:
Ah good catch. It turns out that this buildpack's test runner has two definitions for the `compile()` helper - and they do different things. I'd looked at the one in `utils`, but it turns out that's overridden by the one right at the bottom of `run`:
https://github.com/heroku/heroku-buildpack-apt/blob/4eb4b35d35d0178e5cef73d6998a26ba2c9bbf17/test/utils#L64-L67
https://github.com/heroku/heroku-buildpack-apt/blob/4eb4b35d35d0178e5cef73d6998a26ba2c9bbf17/test/run#L137-L144

For the `compile()` in utils that I'd looked at, the cache dir is set up and torn down as part of the shutils test setup/teardown lifecycle, whereas for the compile function in `run` you have to pass your own cache directory as the second argument in order to test cache re-use.

I've adjusted the new test accordingly, and also split it out to its own testcase so I can also test the other variations like the stack changing as you suggested :-)

Reply:
Diff since last review:
https://github.com/heroku/heroku-buildpack-apt/compare/1436a0bdec606231a143c871a59e97c830a47d63..59d347df11c26bc85865f205245494cc33811f87